File: day_4/task_2.R

total = 0
for (pair in strsplit(readLines("input.txt"), split=",")) {

    el <- strsplit(pair, split="-")

    f <- c(el[[1]][1]:el[[1]][2])
    s <- c(el[[2]][1]:el[[2]][2])

    i <- intersect(f, s)

    total = total + if(length(i)>0) 1 else 0
}

print(paste("Total: ", total))

File: 2024/day_1/task_2.py
with open('input.txt') as f:
    lines = [l.strip().split() for l in f.readlines()]

r = [int(c[1]) for c in lines]
rez = sum([int(c[0])*r.count(int(c[0])) for c in lines])
print(f'Result: {rez}')
